### PR TITLE
Incremental test for MDRangePolicy without using views uptil 4 ranks.

### PR DESCRIPTION
**Pull request by rgayatri23**
Incremental test number 4. MDRangePolicy tests till 4 ranks, without using views.

### CODE REVIEW

**Comment by masterleinad** [comment]
Retest this please.

**Comment by rgayatri23** [comment]
Retest this please.

File: core/unit_test/incremental/Test04_MDRangePolicy.hpp [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+struct MDFunctor {
+  value_type *_data;
+
+  MDFunctor(value_type *data) : _data(data) {}
+
+  // 2D
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i, const int j) const {
+    _data[i * M + j] = i * j * delta;
```
Comment (crtrott) [inline core/unit_test/incremental/Test04_MDRangePolicy.hpp]: This won't catch transpose issues. This needs to be something like `i*100 + j;`

Comment (crtrott) [reply core/unit_test/incremental/Test04_MDRangePolicy.hpp]: But we merge for now and do a big fix at the end.

**Pull request merged**